### PR TITLE
Add global error when user registry is not found

### LINKED ISSUE #2489: Global error when there are no registries available (Ldap,etc) for VMMService
Even if the Ldap configuration is incorrect, we attempt to use Ldap and encounter confusing errors inside of one consistent error message.

For example, if the LdapType is misconfigured, we do log configuration issue at startup, but it isn't exactly clear that :
CWWKG0032W: Unexpected value specified for property [ldapType], value = [badValue]. Expected value(s) are: [Microsoft Active Directory][Custom][IBM Lotus Domino][Novell eDirectory][IBM Tivoli Directory Server][Sun Java System Directory Server][Netscape Directory Server][IBM SecureWay Directory Server].
CWWKG0058E: The element ldapRegistry with the unique identifier ldapregistry is missing the required attribute ldapType.

However, when logins, searches, etc, occur, we try to use the LdapAdapter and strange errors can occur such as NPEs, etc. NPEs can be seen with security/wim trace enabled. For example:

com.ibm.ws.security.registry.RegistryException               > <init> Entry  
                                                                                                               null
                                                                                                               java.lang.NullPointerException
	at com.ibm.ws.security.wim.ProfileManager.searchImpl(ProfileManager.java:1058)
	at com.ibm.ws.security.wim.ProfileManager.genericProfileManagerMethod(ProfileManager.java:229)
	at com.ibm.ws.security.wim.ProfileManager.search(ProfileManager.java:212)
	at com.ibm.ws.security.wim.VMMService.search(VMMService.java:227)
	at com.ibm.ws.security.wim.registry.util.SearchBridge.getUsers(SearchBridge.java:147)
	at com.ibm.ws.security.wim.registry.WIMUserRegistry.getUsers(WIMUserRegistry.java:231)

On a checkPassword request, the user will not be found. Trace will show an underlying NPE.

If the VMMService came up, but no registries can be found, then we could log that (registry count is 0). If there are multiple registries defined, but some aren't available, the VMMService doesn't know which ones are missing, but if we don't have any registries, we know we can't run correctly then.

### PR DESCRIPTION
**Pull request by kristip17**
Fixes #2489

### CODE REVIEW

**Comment by kristip17** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.wim.audit_fat,com.ibm.ws.security.wim.adapter.ldap_fat,com.ibm.ws.security.wim.core_fat,com.ibm.ws.security.wim.registry_fat,com.ibm.ws.security.wim.scim_fat

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 1]
```diff
@@ -68,6 +68,10 @@ FEDERATED_MANAGER_SERVICE_STOPPED=CWIMK0010I: The user registry federation servi
 FEDERATED_MANAGER_SERVICE_STOPPED.explanation=This message is for informational purposes only.
 FEDERATED_MANAGER_SERVICE_STOPPED.useraction=No action is required.
 
+MISSING_REGISTRY_DEFINITION=CWIMK0011E: The user registry operation could not be completed. A valid user registry was not found. Add or correct the registry definition in the server.xml file.
+MISSING_REGISTRY_DEFINITION.explanation=A user registry was not defined or could not be initialized due to a configuration error. A valid user registry includes an LDAP registry or a custom registry.
+MISSING_REGISTRY_DEFINITION.useraction=If a user registry was defined in the configuration file, review the logs for configuration errors during server start up. Add or correct the user registry definition in the server.xml file.
+
```
Comment (rhpeters) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops]: ID reviewed

**Comment by kristip17** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.wim.audit_fat,com.ibm.ws.security.wim.adapter.ldap_fat,com.ibm.ws.security.wim.core_fat,com.ibm.ws.security.wim.registry_fat,com.ibm.ws.security.wim.scim_fat

**Comment by kristip17** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.wim.audit_fat,com.ibm.ws.security.wim.adapter.ldap_fat,com.ibm.ws.security.wim.core_fat,com.ibm.ws.security.wim.registry_fat,com.ibm.ws.security.wim.scim_fat

**Comment by kristip17** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.wim.audit_fat,com.ibm.ws.security.wim.adapter.ldap_fat,com.ibm.ws.security.wim.core_fat,com.ibm.ws.security.wim.registry_fat,com.ibm.ws.security.wim.scim_fat

**Comment by kristip17** [comment]
@rhpeters  -- I changed my new message a little bit -- do you want to re-review?

File: dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops [hunk 2]
```diff
@@ -68,6 +68,10 @@ FEDERATED_MANAGER_SERVICE_STOPPED=CWIMK0010I: The user registry federation servi
 FEDERATED_MANAGER_SERVICE_STOPPED.explanation=This message is for informational purposes only.
 FEDERATED_MANAGER_SERVICE_STOPPED.useraction=No action is required.
 
+MISSING_REGISTRY_DEFINITION=CWIMK0011E: The user registry operation could not be completed. A valid user registry was not found. Add or correct the registry definition in the server.xml file.
+MISSING_REGISTRY_DEFINITION.explanation=A user registry was not defined or could not be initialized due to a configuration error. A valid user registry includes an LDAP registry or other supported registries or repositories.
+MISSING_REGISTRY_DEFINITION.useraction=If a user registry was defined in the configuration file, review the logs for configuration errors during server start up. Add or correct the user registry definition in the server.xml file.
+
```
Comment (rhpeters) [inline dev/com.ibm.ws.security.wim.core/resources/com/ibm/ws/security/wim/util/resources/WimUtilMessages.nlsprops]: ID reviewed

**Comment by Azquelt** [comment]
I think this broke the build because the version of `com.ibm.websphere.appserver.spi.federatedRepository` was updated, but the reference to the javadoc zip generated from this bundle was not updated in the `com.ibm.websphere.appserver.federatedRegistry-1.0.feature` file.

I'm not sure why this didn't cause the personal build to fail.

**Comment by kristip17** [comment]
Re-delivered in #4244

**Pull request merged**